test/conformance/api/v1/service_test.go

```diff
@@ -581,3 +582,61 @@ func TestAnnotationPropagation(t *testing.T) {
 		t.Error("Route still has `juicy` annotation")
 	}
 }
+
+// TestServiceCreateWithMultipleContainers tests both Creation paths for a service.
+// The test performs a series of Validate steps to ensure that the service transitions as expected during each step.
+func TestServiceCreateWithMultipleContainers(t *testing.T) {
+	t.Parallel()
+	clients := test.Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+	}
+
+	// Clean up on test failure or interrupt
+	test.EnsureTearDown(t, clients, &names)
+	// images are used to validate digest in validateControlPlane function
+	images := map[string]string{
+		"serving-container": test.ServingContainer,
+		"sidecar-container": test.SidecarContainer,
+	}
+	containers := []corev1.Container{{
+		Name:  "serving-container",
+		Image: pkgTest.ImagePath(test.ServingContainer),
+		Ports: []corev1.ContainerPort{{
+			ContainerPort: 8881,
+		}},
+	}, {
+		Name:  "sidecar-container",
+		Image: pkgTest.ImagePath(test.SidecarContainer),
+	}}
+
+	// Setup initial Service
+	objects, err := v1test.CreateServiceReadyForMultiContainer(t, clients, &names, func(svc *v1.Service) {
```

Comment:
if I understand correctly, the only difference between `CreateServiceReady` and `CreateServiceReadyForMultiContainer` is that the latter doesn't check that names.Image is set? If so I'm wondering if instead of using the absence of the names.Image to go to the multi-container path in ValidateImageDigest we can just use the presence of the map? That way we can drop CreateServiceReadyForMultiContainer. 

I'm also sort of wondering why we don't just add a names.Images for the multi-container case, but maybe there's a good reason not to do that?

Reply:
The only reason to add extra function for multicontainer `CreateServiceReadyForMultiContainer` in order to avoid lot of code changes(multiple files)

1. Yes that is one point
The other thing is names.Images is `string` but for multicontainer there will be more than one containers so need to change type of `names.Images` to `[]string` I din't change initially because it requires changes in lot of places.

2. The another difference is for single container we use this https://github.com/knative/serving/blob/master/test/v1/service.go#L234-L236 and by passing names.Images we can able to create configuration but for multicontainer it will be difficult .

But this is something we can solve and make use of  `CreateServiceReady` but there will be lot of changes in multiple files if its okay then i can work on that.

Reply:
I think you don't need changes in multiple files, fwiw. Just set names.Image in this function and the existing CreateServiceReady will work, without modification (so no effect elsewhere). You can then use *either* the presence of the optional `imagesForMulticontainer` parameter, *or* names.Images being non-empty (if we decide to add that) to decide whether to test multiple digests or not. 

I can't immediately think of a reason not to allow a map or []string in `names` so we could have names.Images,  @vagababov do you have any opinions on that? (I think for this PR we'd either just add an additional Images field, so no need for lots of changes, the existing Image field would stay, then maybe refactor; or we could refactor to names.Image->names.Images in another PR first and then rebase this on top).

Reply:
I agree with the refactor potential, though this PR seems only tangentially affected at this point. I'd be fine refactoring after a merge too, if still necessary.